scripts/register_parachain.sh

```diff
@@ -11,46 +11,40 @@ seed=$2
 wasm=$3
 genesis=$4
 parachain_id=$5
-tokens=$6
-account=$7
+types=$6 # we can remove this once parachain types are included in polkadot-js-api
 
 [ -z "$url" ] && usage
 [ -z "$seed" ] && usage
 [ -z "$wasm" ] && usage
+[ -z "$types" ] && usage
 [ -z "$genesis" ] && usage
 [ -z "$parachain_id" ] && usage
-[ -z "$tokens" ] && usage
-[ -z "$account" ] && usage
 if ! [ -r "$wasm" ]; then
     echo "Could not read: $wasm"
     exit 1
 fi
+if ! [ -r "$types" ]; then
+    echo "Could not read: $types"
+    exit 1
+fi
 
 if ! which polkadot-js-api &> /dev/null; then
     echo 'command `polkadot-js-api` not in PATH'
-    echo "npm install -g @polkadot/api-cli"
+    echo "npm install -g @polkadot/api-cli@beta"
     exit 1
 fi
 
 set -e -x
 
 test -f "$seed" && seed="$(cat "$seed")"
 
-polkadot-js-api \
-    --ws "${url?}" \
-    --sudo \
-    --seed "${seed?}" \
-    tx.registrar.registerPara \
-        "${parachain_id?}" \
-        '{"scheduling":"Always"}' \
-        @"${wasm?}" \
-        "${genesis?}"
+wasm=$(cat $wasm)
 
 polkadot-js-api \
     --ws "${url?}" \
     --sudo \
     --seed "${seed?}" \
-    tx.balances.setBalance \
-        "${account?}" \
-        $((tokens * 10 ** 12)) \
-        0
+    --types "${types?}" \
+    tx.parasSudoWrapper.sudoScheduleParaInitialize \
+        "${parachain_id?}" \
+        "{ \"genesisHead\":\"${genesis?}\", \"validationCode\":\"${wasm?}\", \"parachain\": true }" \
```

Comment:
Doesn't this put the entire ~1MB wasm blob on the CLI? Was your shell able to handle that?

Reply:
Shell should be able to handle 1MB easily. Many terminal emulators might not. Just send output of the script to `/dev/null` if it's a problem.

Reply:
I try 
./register.sh ws://127.0.0.1:9944 "//Alice" ~/genesis.wasm $(cat ~/genesis-state) 14 temp_parachain_types.json > /dev/null 2>&1
but nothing happened in my network